modin/engines/base/frame/data.py

```diff
@@ -1741,6 +1741,13 @@ def to_pandas(self):
             )
             df.index = self.index
             df.columns = self.columns
+
+        # Don't expose the internal implementation detail of __reduced__ to
+        # Pandas:
+        if df.index.name == "__reduced__":
```

Comment:
I prefer to fix this internally when the groupby occurs (or before).

Reply:
Done, although it could potentially be done even deeper.